crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/CheckDiagnosticsCommand.java

```diff
@@ -66,18 +66,19 @@ public CallStatus run() {
                     String errorDetails = entry.getKey().toString() + " " + entry.getValue().getDetails();
                     Error.push("diagnosticCheck failed! " + errorDetails);
                     LOGGER.debug("diagnosticCheck failed!", errorDetails);
-                    return new CallStatus(Error.get(RestCrudConstants.ERR_REST_CHECK_DIAGNOSTICS, errorDetails));
+                    healthCheckResults.put(entry.getKey(), entry.getValue());
+                    healthCheckResponse = Response.Status.SERVICE_UNAVAILABLE;
```

Comment:
Also, just double checking: so the verdict is all services' /diagnostics should return 503 when any check is unhealthy just like /health?

Reply:
Yeah I briefly considered making diagnostics always return 200 (even if there is a failure), and health return 200 or 500 if it is working/failing, but was swayed by the following sentence in the 503 usage description:

> The implication is that this is a temporary condition which will be alleviated after some delay

So for consistency sake and to have one less mind map we have to maintain, both health and diagnostics will either return 200 when everything is passing or 503 when one or more things are failing.  If we see anything else (like a 500 or 404 or something), it is a problem with the deployment.  So whatever is checking these things (monitoring or load balancers) can rely on 200 response to indicate everything is okay, and anything else (503, 500, 404, etc) means there is a problem (the details of which should be reflected in the endpoint response).

Reply:
> Looks good, just need the health check details added to the result and I think this is good to go.

This line 
> healthCheckResults.put(entry.getKey(), entry.getValue());
will add the details too since they are part of the HealthCheck.Result object, however I needed to add something in ReadHealthCheck to populate the details with the output of the test read (which is now done).